lib/server/services/log/LogService.js

```diff
@@ -47,6 +47,44 @@ class LogService {
         return log ? logAdapter.toEntity(log) : null;
     }
 
+    /**
+     * Retrieves a paginated list of all root logs along with the total count.
+     * @param {number} limit The maximum number of root logs to retrieve in a single call.
+     * @param {number} offset The number of root logs to skip before starting to return results.
+     * @return {Promise<{count: number, logs: Log[]}>} A promise that resolves with an object containing the total count of all root logs,
+     * and an array of the root logs for the current page, or null if no logs are found.
+     */
+    async getAllRootLogs(limit, offset) {
```

Comment:
Please add a few tests for this function

Reply:
done